Makefile

```diff
@@ -183,7 +183,7 @@ fix-format:
 # Run all checks in order of speed / likely failure.
 # As a last thing, run attribute registry generation and git-diff for differences.
 .PHONY: check
-check: misspell markdownlint check-format markdown-toc compatibility-check markdown-link-check check-policies attribute-registry-generation
+check: misspell markdownlint check-format markdown-toc markdown-link-check check-policies attribute-registry-generation
```

Comment:
should we remove `compatibility-check` from Makefile?

Reply:
it's based on the build-tools check and it's duplicating weaver-based backcompat checks in rego files.

We keep it as optional CI check and it helps find some missing policies like https://github.com/open-telemetry/semantic-conventions/issues/1686 or gaps in weaver. 

I don't think anyone needs to run it on their machine though